src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -608,6 +608,12 @@ Dependencies DatabaseCatalog::getDependencies(const StorageID & from) const
         return {};
     return Dependencies(iter->second.begin(), iter->second.end());
 }
+ 
+void DatabaseCatalog::getViewDependencies(ViewDependencies & view_dependencies_) const
+{
+    std::lock_guard lock{databases_mutex};
+    view_dependencies_ = view_dependencies;
```

Comment:
It is not thread safe interface, we should copy view dependencies to client.
Better API will be:
```
ViewDependencies DatabaseCatalog::getViewDependencies()
```